main.go

```diff
@@ -213,15 +221,111 @@ func fetchOrCreateTestPlan(ctx context.Context, cfg config.Config, files []strin
 		}
 		// Create the fallback plan
 		fmt.Println("Could not fetch plan from server, using fallback mode. Your build may take longer than usual.")
-		testPlan = plan.CreateFallbackPlan(testCases, cfg.Parallelism)
+		testPlan = plan.CreateFallbackPlan(files, cfg.Parallelism)
 	}
 
 	// The server can return an "error" plan indicated by an empty task list (i.e. `{"tasks": {}}`).
 	// In this case, we should create a fallback plan.
 	if len(testPlan.Tasks) == 0 {
 		fmt.Println("Test splitter server returned an error, using fallback mode. Your build may take longer than usual.")
-		testPlan = plan.CreateFallbackPlan(testCases, cfg.Parallelism)
+		testPlan = plan.CreateFallbackPlan(files, cfg.Parallelism)
 	}
 
 	return testPlan, nil
 }
+
+type fileTiming struct {
+	Path     string
+	Duration time.Duration
+}
+
+// createRequestParam creates the request parameters for the test plan.
+// If SplitByExample is disabled (default), it will return the default params that contain all the files.
```

Comment:
The include/exclude filters have already been applied at this point right? Might be worth a mention